alexandria/core/validations.py

```diff
@@ -62,17 +64,40 @@ def validate_file(self, data, context):
         validate_file_infection(data["content"])
 
         # Validate that the mime type is allowed in the category
-        mime_type = data["content"].content_type
-        if mime_type == "application/octet-stream" or not mime_type:
-            guess, encoding = guess_type(data["name"])
-            if guess is not None:
-                mime_type = guess
-            else:
-                data["content"].seek(0)
-                mime_type = magic.from_buffer(data["content"].read(), mime=True)
-                data["content"].seek(0)
-
-        validate_mime_type(mime_type, data["document"].category)
-        data["mime_type"] = mime_type
+        content_type_header = data["content"].content_type
+        extension_type, _ = guess_type(data["name"])
+
+        if not content_type_header:  # pragma: no cover
+            raise ValidationError(gettext_lazy("Content-Type header is missing."))
+        if not extension_type:
+            raise ValidationError(gettext_lazy("The uploaded file has no extension."))
```

Comment:
Not exactly true: If I understand correctly, the meaning is that the file name extension could not be mapped to a known mime type

Reply:
True, fixed the wording